packages/kbn-typed-react-router-config/src/use_params.ts

```diff
@@ -23,9 +23,10 @@ export function useParams(...args: any[]) {
     args.pop();
   }
 
-  const paths = args as string[];
-
-  const getParamsArgs = [...paths, location, optional] as [never, Location<any>, boolean];
-
-  return router.getParams(...getParamsArgs);
+  return useMemo(() => {
+    // @ts-expect-error
+    return router.getParams(...args, location, optional);
+    // it is not possible to statically verify the dependencies when spreading args.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router, ...args, location, optional]);
```

Comment:
@dgieselaar I need a review from the knowledge team apparently. Mind taking a look? Thanks